syntax = "proto3";
package com.registry.grpc;

message RegistryRequest {    
    int32 ID=1;
    string name=2;
    string password=3;
}

message RegistryResponse{
    string response=1;
}

message loginRequest{
    string username=1;
    string password=2;
}

message changeUserInfo{
    string username = 1;
    string password = 2;
    string newUsername = 3;
    string newPassword = 4;
    
}

service RegistryService{
    rpc Registry(RegistryRequest) returns (RegistryResponse);
}

service login{
    rpc Login(loginRequest) returns (RegistryResponse);
}

service modifyUser{
    rpc Modify(changeUserInfo) returns (RegistryResponse);
}